src/Writer/CsvWriter.php

```diff
@@ -66,7 +66,7 @@ class CsvWriter implements TypedWriterInterface
      * @param bool   $showHeaders
      * @param bool   $withBom
      */
-    public function __construct($filename, $delimiter = ',', $enclosure = '"', $escape = '\\', $showHeaders = true, $withBom = false)
+    public function __construct(string $filename, string $delimiter = ',', string $enclosure = '"', string $escape = '\\', bool $showHeaders = true, bool $withBom = false)
```

Comment:
Told you things would get ugly when adding type hinting :P Not using multiline signatures really grinds my gears.

Reply:
Maybe we should find an other way if we have more than 4(?) parameters for a method

Reply:
An option resolver might be it